### ISSUE TITLE
Unable to nullify choice fields via the API

### ISSUE THREAD

**Issue by fantasmita** [netbox-community/netbox#4083]
### Environment
* Python version:  3.6.5
* NetBox version:  2.7.3 

To fully unrack a device, you must blank out its rack, position, and face values.
The `face` field has a non-null constraint and has an API validator that requires its value to be either `front` or `rear`, so it's not possible to blank out the value through the API which prevents the device from being removed from the rack.

### Steps to Reproduce
1. Create a device in netbox with values for rack, position, and face.
2. Try to clear all three values with a PATCH like the following
```
curl -X PATCH "https://$NETBOX/api/dcim/devices/$ID/" -H  "accept: application/json" -H  "Content-Type: application/json" -H  "X-CSRFToken: $TOKEN" -d "{  \"rack\": null,  \"position\": null,  \"face\": \"\"}"
```

### Expected Behavior
The device to be unracked

### Observed Behavior
```
{
  "face": [
    " is not a valid choice."
  ]
}
```

If passing `null` for the face value, the django non-null constraint is hit.
If patching just the `rack` and `position` values without touching the `face`, the `Cannot select a rack face without assigning a rack.` validator is hit.

**Comment by jeremystretch** [comment]
Omit the `face` field if you are unracking a device: It will be nullified automatically. The following request is successful:

```
curl -X PATCH http://localhost:8000/api/dcim/devices/<ID>/ \
-H "Authorization: Token <TOKEN>" \
-H "Content-Type: application/json" \
-H "Accept: application/json; indent=4" \
-d '{"rack": null, "position": null, "face": ""}'
```

**Comment by fantasmita** [comment]
Thanks for looking and for the suggestions.
However, I don't think the nullification is working as expected.

I know netboxdemo.com is unsanctioned, but here's a quick reproducible illustration of the problem in that environment when following your suggestions.


setting `face` to the empty string:
```
$ curl -X PATCH "https://netboxdemo.com/api/dcim/devices/1/" \
>     -H "Authorization: Token  72830d67beff4ae178b94d8f781842408df8069d" \
>     -H "Content-Type: application/json" \
>     -H "Accept: application/json; indent=4" \
>     -d '{"rack": null, "position": null, "face": ""}'
{
    "face": [
        " is not a valid choice."
    ]
}
```
omitting `face`:
```
$ curl -X PATCH "https://netboxdemo.com/api/dcim/devices/1/" \
>     -H "Authorization: Token  72830d67beff4ae178b94d8f781842408df8069d" \
>     -H "Content-Type: application/json" \
>     -H "Accept: application/json; indent=4" \
>     -d '{"rack": null, "position": null}'
{
    "face": [
        "Cannot select a rack face without assigning a rack."
    ]
}
```

**Comment by jeremystretch** [comment]
Sorry, I had pasted the wrong command in my previous comment. I've corrected it to omit the `face` field. This is working on v2.7.4.

**Comment by fantasmita** [comment]
No worries, thanks.

I'm sorry for not explaining myself more clearly, but omitting the `face` when making a PATCH request doesn't successfully nullify the value before it hits the site / rack validators, which results in the `Cannot select a rack face without assigning a rack.` validation error I pasted above.
I confirmed today that this still occurs on a fresh installation of v2.7.4.

Anyway, I'm happy to work around this and it's definitely possible I'm missing something, but let me know if you have other thoughts or if I can provide any other information. Thanks again for your time
